Sources/KeyValueStorage/KeyValueStorageObserving/KeyValueStorageObserving+mocks.swift

```diff
@@ -0,0 +1,49 @@
+#if DEBUG
+import Combine
+import Foundation
+
+// MARK: - Stub
+
+public class KeyValueStorageObservingStub: KeyValueStorageObservable {
+  private let observePublisher: AnyPublisher<Decodable?, KeyValueStorageError>
+
+  public init(
+    observe: AnyPublisher<Decodable?, KeyValueStorageError> = Just(nil).setFailureType(to: KeyValueStorageError.self).eraseToAnyPublisher()
+  ) {
+    self.observePublisher = observe
+  }
+
+  public func observe<T>(
+    forKey: KeyProviding,
+    jsonDecoder: JSONDecoder
+  ) -> AnyPublisher<T?, KeyValueStorageError> where T: Decodable {
+    observePublisher
+      .map { $0 as? T }
+      .eraseToAnyPublisher()
+  }
```

Comment:
možno by sme mohli z hľadiska lepšej čitelnosti ukázať, že sa parametre v tejto funkci vôbec nepoužívajú
```swift
public func observe<T>(
    forKey _: KeyProviding,
    jsonDecoder _: JSONDecoder
)
```
čo si o tom myslíš @troupmar? Vzťahovalo by sa to k viacerým funkcím z tohto MR.